src/shims/unix/fs.rs

```diff
@@ -1197,6 +1197,65 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
         }
     }
 
+    fn posix_fallocate(
+        &mut self,
+        fd_num: i32,
+        offset: i128,
+        len: i128,
+    ) -> InterpResult<'tcx, Scalar> {
+        let this = self.eval_context_mut();
+
+        // According to the man page of `possix_fallocate`, it returns the error code instead
+        // of setting `errno`.
+        let ebadf = Scalar::from_i32(this.eval_libc_i32("EBADF"));
+        let einval = Scalar::from_i32(this.eval_libc_i32("EINVAL"));
+        let enodev = Scalar::from_i32(this.eval_libc_i32("ENODEV"));
```

Comment:
I think it's better to just inline these where they are needed; the error path shouldn't take so much space on the happy path.

Also, you can use `eval_libc` instead of `Scalar::from_i32(eval_libc_i32)`.